src/sources/static_metrics.rs

```diff
@@ -58,14 +58,21 @@ impl Default for StaticMetricsConfig {
 
 /// Tag configuration for the `internal_metrics` source.
 #[configurable_component]
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug)]
 #[serde(deny_unknown_fields, default)]
 pub struct StaticMetricConfig {
     /// Name of the static metric
     pub name: String,
 
     /// "Observed" value of the static metric
-    pub value: f64,
+    #[configurable(derived)]
+    #[serde(default = "default_metric_value")]
```

Comment:
Is a default value necessary here? I think we should fail if the user does not specify any value. In which cases would the default value be useful?